pipeline/authz/keto_engine_acp_ory_test.go

```diff
@@ -117,14 +99,8 @@ func TestAuthorizerKetoWarden(t *testing.T) {
 			expectErr: true,
 		},
 		{
-			config: []byte(`{ "required_action": "action:$1:$2", "required_resource": "resource:$1:$2" }`),
-			rule: &rule.Rule{
-				Match: &rule.Match{
-					Methods: []string{"POST"},
-					URL:     "https://localhost/api/users/<[0-9]+>/<[a-z]+>",
-				},
-			},
-			r: &http.Request{URL: urlx.ParseOrPanic("https://localhost/api/users/1234/abcde")},
+			config: []byte(`{ "required_action": "action:{{ index .MatchContext.RegexpCaptureGroups 0}}:{{ index .MatchContext.RegexpCaptureGroups 1}}", "required_resource": "resource:{{ index .MatchContext.RegexpCaptureGroups 0}}:{{ index .MatchContext.RegexpCaptureGroups 1}}" }`),
```

Comment:
This is a breaking change, right?

Reply:
Yes it is.  Do we need to keep the compatibility ?

Reply:
Yes, did you see my overall comment?

https://github.com/ory/oathkeeper/pull/358#pullrequestreview-358820288

Hope this helps. If you need guidance let me know

Reply:
Oops, I didn't see it.

I'll add that. 
I just need to figure out how to replace the "$**x**" with "{{printIndex .MatchContext **x-1**}}"

Reply:
Without the index offset, I could write that:

```go
re := regexp.MustCompile(`\$([0-9]+)`)
result := re.ReplaceAllString("my:action:$2:foo:$1", "{{ printIndex .MatchContext.RegexpCaptureGroups $1}}")
```

Reply:
You could write an increment/decrement function (or use one from sprig if it exists) to increment/decrement found indices by one. Alternatively yes, you could also go back to having the 0 index as the full regex result.

We definitely should document this :)